### PR TITLE
Reland integration template

### PR DESCRIPTION
**Pull request by dnfield**
Must not be landed until after https://github.com/flutter/flutter/pull/71738

Otherwise this is an identical reland. The linked PR fixes a bug in our Android build logic to make this work now.

/cc @blasten @jonahwilliams @Piinks

### CODE REVIEW

File: packages/flutter_tools/test/integration.shard/integration_test_test.dart [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+      ],
+      workingDirectory: projectDir.path,
+    );
+
+    if (result.exitCode != 0) {
+      print('================================= STDOUT =======================================');
+      print(result.stdout);
+      print('================================= STDERR =======================================');
+      print(result.stderr);
+      fail('flutter drive failed, see output.');
```
Comment (MrBirb) [inline packages/flutter_tools/test/integration.shard/integration_test_test.dart:59]: This is probably flutter driver* right?

Comment (dnfield) [reply packages/flutter_tools/test/integration.shard/integration_test_test.dart:59]: No, this is referring to the command `flutter drive` (which is unfortunately easily confused with and definitely related to flutter_driver).

Comment (MrBirb) [reply packages/flutter_tools/test/integration.shard/integration_test_test.dart:59]: Ah so sorry!

**Comment by dnfield** [comment]
The PR that fixed functionality this depends on has landed.

File: packages/flutter_tools/test/integration.shard/integration_test_test.dart (hunk 1 above)
Comment (dnfield) [reply packages/flutter_tools/test/integration.shard/integration_test_test.dart:59]: No worries, thanks for looking!

**Review by blasten — APPROVED** [review]
LGTM!

**Comment by dnfield** [comment]
I'm going to file a follow up PR to address updating the default iOS project to support running a test out of the box without flutter drive.

**Pull request merged**